### PR TITLE
Fixes extreme lag caused by bots crossing lava.

### PR DESCRIPTION
**Pull request by Time-Green**
Fixes: https://github.com/yogstation13/yogstation/issues/703

:cl:
fix: Fixes a bug that would cause immense lag and even crashing when a bot would burn up in lava.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/effect_system/effects_explosion.dm [hunk 1]
```diff
@@ -54,4 +54,9 @@
 	spawn(5)
 		var/datum/effect_system/smoke_spread/S = new
 		S.set_up(2, location)
-		S.start()
\ No newline at end of file
+		S.start()
+
+/obj/effect/explosion/alt
```
Comment (george99g) [inline code/game/objects/effects/effect_system/effects_explosion.dm]: Couldn't you have used the already-existing explosion effect code to implement this?

Comment (Time-Green) [reply code/game/objects/effects/effect_system/effects_explosion.dm]: Believe me, this one is cooler
I am trying to get a vid showing it up.
Expect it in half an hour, maybe.

Comment (george99g) [reply code/game/objects/effects/effect_system/effects_explosion.dm]: Well then. Why not name it appropriately?
"alt" is not a great type name that'll help people understand how your specific type is different.

Comment (Time-Green) [reply code/game/objects/effects/effect_system/effects_explosion.dm]: I'll call it mushroom then. Since it looks like a mushroom

Comment (george99g) [reply code/game/objects/effects/effect_system/effects_explosion.dm]: :+1:

Comment (Time-Green) [reply code/game/objects/effects/effect_system/effects_explosion.dm]: Forget the video, I can't get that to work.

**Comment by Time-Green** [comment]
Alright, merge ready.

File: code/game/turfs/simulated/floor/plating.dm [hunk 2]
```diff
@@ -239,6 +239,13 @@
 /turf/open/floor/plating/lava/proc/burn_stuff()
 	. = 0
 	for(var/thing in contents)
+		if(istype(thing, /mob))
```
Comment (Super3222) [inline code/game/turfs/simulated/floor/plating.dm:242]: This could've been placed before the regions of `istype(thing, /mob/living))` you know.

**Comment by Cruix** [comment]
Post a gif or a video of the explosion and I might merge this anyway.

**Comment by Time-Green** [comment]
I tried but I can't record stuff. Doeesn't matter anyway.

**Pull request closed without merging**